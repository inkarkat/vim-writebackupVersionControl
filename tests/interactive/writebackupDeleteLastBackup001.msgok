Test: No backups exist for this file
No backups exist for this file.

User: PLEASE PRESS 'No'
Really delete backup '20080101b'?
[N]o, (Y)es: 
Delete canceled.

User: PLEASE PRESS 'Yes'
Really delete backup '20080101b'?
[N]o, (Y)es: 
Deleted backup '20080101b'; last backup now is '20080101a'

Deleted backup '20080101a'; last backup now is '20061231a'

Deleted backup '20061231a'; last backup now is '20061201a'
