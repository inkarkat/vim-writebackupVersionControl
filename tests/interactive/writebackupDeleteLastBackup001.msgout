
Executing shell command: setup
Archive:  /home/inkarkat/Unixhome/.vim/tests/WriteBackup/testdata.zip
   creating: /tmp/WriteBackupTest/another dir/
  inflating: /tmp/WriteBackupTest/another dir/some file.txt  
   creating: /tmp/WriteBackupTest/backup/
  inflating: /tmp/WriteBackupTest/backup/someplace else.txt.20080124a  
  inflating: /tmp/WriteBackupTest/backup/someplace else.txt.20080124b  
  inflating: /tmp/WriteBackupTest/backup/someplace else.txt.20080124c  
  inflating: /tmp/WriteBackupTest/important.txt  
  inflating: /tmp/WriteBackupTest/important.txt.19990815a  
  inflating: /tmp/WriteBackupTest/important.txt.20061201a  
  inflating: /tmp/WriteBackupTest/important.txt.20061231a  
  inflating: /tmp/WriteBackupTest/important.txt.20080101a  
  inflating: /tmp/WriteBackupTest/important.txt.20080101b  
  inflating: /tmp/WriteBackupTest/not important.txt  
  inflating: /tmp/WriteBackupTest/someplace else.txt  

"not important.txt" 
"not important.txt" 1L, 10C
"important.txt" 
"important.txt" 2L, 28C
User: PLEASE PRESS 'No'
Really delete backup '20080101b'?
[N]o, (Y)es: 
Delete canceled.
User: PLEASE PRESS 'Yes'
Really delete backup '20080101b'?
[N]o, (Y)es: 
Deleted backup '20080101b'; last backup now is '20080101a'
Deleted backup '20080101a'; last backup now is '20061231a'
Deleted backup '20061231a'; last backup now is '20061201a'
24 more lines
<ixhome/.vim/tests/WriteBackup/interactive/writebackupDeleteLastBackup001.out" 
<Backup/interactive/writebackupDeleteLastBackup001.out" [New] 25L, 498C written
