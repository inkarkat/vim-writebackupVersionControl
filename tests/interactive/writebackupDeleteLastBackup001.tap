1..3
ok 1 - error shown
ok 2 - last backup still exists
ok 3 - last backup was removed
