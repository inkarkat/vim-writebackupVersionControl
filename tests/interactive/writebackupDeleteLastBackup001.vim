" Test deletion of last backup.

if has('gui_running')
    call vimtest#SkipMsgout('Dialog confirmation message is not captured in GUI.')
endif

call vimtest#StartTap()
call vimtap#Plan(3)

cd $TEMP/WriteBackupTest
edit not\ important.txt

call vimtap#err#Errors('No backups exist for this file.', 'WriteBackupDeleteLastBackup', 'error shown')

edit important.txt
call vimtest#RequestInput('No')
WriteBackupDeleteLastBackup
call vimtap#Ok(filereadable('important.txt.20080101b'), 'last backup still exists')
call vimtest#RequestInput('Yes')
WriteBackupDeleteLastBackup
call vimtap#Ok(! filereadable('important.txt.20080101b'), 'last backup was removed')
WriteBackupDeleteLastBackup!
WriteBackupDeleteLastBackup!

call ListFiles()
call vimtest#Quit()
