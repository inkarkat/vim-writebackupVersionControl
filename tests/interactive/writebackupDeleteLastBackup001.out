.
    [DIRECTORY]
./another dir
    [DIRECTORY]
./another dir/some file.txt
    -> just junk
./backup
    [DIRECTORY]
./backup/someplace else.txt.20080124a
    -> here
./backup/someplace else.txt.20080124b
    -> there
./backup/someplace else.txt.20080124c
    -> and everywhere
./important.txt
    -> current revision
./important.txt.19990815a
    -> it all started here
./important.txt.20061201a
    -> first revision
./not important.txt
    -> just junk
./someplace else.txt
    -> it's a song

